src/client/Microsoft.Identity.Client/Credential/KeyMaterialInfo.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Identity.Client.Credential
+{
+    /// <summary>
+    /// Class to store crypto key information for a Managed Identity supported Azure resource.
+    /// For more details see https://aka.ms/msal-net-managed-identity
+    /// </summary>
+    internal class KeyMaterialInfo
+    {
+        private const string IsKeyGuardEnabledProperty = "Virtual Iso";
+        private static bool _isPopSupported = false;
+        internal static readonly string s_credentialEndpoint = "http://169.254.169.254/metadata/identity/credential?cred-api-version=1.0";
+        private static CryptoKeyType s_cryptoKeyType = CryptoKeyType.None;
+        internal const string KeyProviderName = "Microsoft Software Key Storage Provider";
+        internal const string KeyName = "ManagedIdentityCredentialKey";
+        private readonly ECDsaCng _eCDsaCngKey;
+        private readonly RSA _rsaKey;
+
+        public KeyMaterialInfo(bool clientCapabilitiesRequested)
+        {
+            _eCDsaCngKey = GetMachineKey(KeyProviderName, KeyName);
+            
+            if (clientCapabilitiesRequested && _eCDsaCngKey == null)
+            {
+                _rsaKey = CreateRsaKey();
+            }
+        }
+
+        public static bool IsPoPSupported => _isPopSupported;
```

Comment:
SRP again. 

The "KeyMaterialInfo" class deals with keys and so its properties names should be in accordance with the key domain. Things like "IsHardwareKey", "IsSoftwareKey" etc. 

Some other class should be responsible for translating key capabilities into MSAL capabiltiies.